msdk-io/msdk-io-netcdf/src/main/java/io/github/msdk/io/netcdf/NetCDFMsScan.java

```diff
@@ -54,51 +54,70 @@ public NetCDFMsScan(Integer scanNumber, int[] scanStartPositions, float[] scanRe
 
   @Override
   public float[] getIntensityValues(float[] intensityValues) {
-    final Integer scanIndex = getScanIndex();
-    numOfDataPoints = getNumberOfDataPoints();
-    try {
-      final int scanStartPosition[] = {scanStartPositions[scanIndex]};
-      final int scanLength[] = {scanStartPositions[scanIndex + 1] - scanStartPositions[scanIndex]};
-      final Array intensityValueArray = intensityValueVariable.read(scanStartPosition, scanLength);
-      final Index intensityValuesIndex = intensityValueArray.getIndex();
-
+    if (this.intensityValues == null) {
```

Comment:
maybe better to rename this.intensityValues to something like this.preloadedIntensityValues, for clarity